### ISSUE TITLE
deploytime-exporter build crashing

### ISSUE THREAD

**Issue by GuilhermeCamposo** [konveyor/pelorus#303]
deploytime-exporter Buildconfig is using registry.access.redhat.com/ubi8/python-38, but source code is requiring Python >= 3.9.0

**Comment by rmarting** [comment]
I got the same issue testing in my local CRC instance.

Builder output log:

```text
# Omitted output
Storing signatures
Generating dockerfile with builder image registry.access.redhat.com/ubi8/python-38
STEP 1: FROM registry.access.redhat.com/ubi8/python-38
# Omitted output
---> Installing application ...
Processing /opt/app-root/src
ERROR: Package 'pelorus' requires a different Python: 3.8.6 not in '>=3.9'
WARNING: You are using pip version 19.3.1; however, version 21.1.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
error: build error: error building at STEP "RUN /usr/libexec/s2i/assemble": error while running runtime: exit status 1
```

**Comment by rmarting** [comment]
It seems that this issue is fixed in `master` branch. I got the issue using the 1.3.0.1 tag version. I could be great to release a minor version with this fix.

**Comment by etsauer** [comment]
@KevinMGranger when you get a chance could you cut a 1.3.0.2?

**Comment by KevinMGranger** [comment]
I'll be making a new release soon, but I want to verify-- are you sure you're using the 1.3.0.1 tag @rmarting ? `setup.py` isn't in that tag, so it shouldn't be able to require `>=3.9`.

It's possible you're using the 1.3.0.1 tag for the helm charts, but the exporters are pointing to master in the `values` file. Can you verify what you're seeing for each `exporters.instances[*].source_ref`?

This does bring up a good point: that we should reconsider having the docs and defaults pointing at master.

**Comment by rmarting** [comment]
@KevinMGranger I downloaded the 1.3.0.1 version from [here](https://github.com/konveyor/pelorus/releases/tag/1.3.0.1). However the issue is caused by the definition of a DeploymentConfig of the exporters.

This [line](https://github.com/konveyor/pelorus/blob/1.3.0.1/charts/pelorus/charts/exporters/templates/_buildconfig.yaml#L27) defined the `python-38` image. So this is the root cause of the issue. That line in `master` is defined as `python-39` and works successfully.

**Comment by etsauer** [comment]
@KevinMGranger agree with your last point. This issue points out a flaw in our release process. What good is a release tag is the builds inside of it (or image tags if we ever get around to #63 ) still point to master? This seems like a problem that someone in GitHub land would have solved before...

**Comment by KevinMGranger** [comment]
@etsauer Some ways I can think of to solve this:

1. A "stable" branch that lags behind master, and make that the default source_ref in the charts. This is probably the simplest solution, and only requires a small change to the release candidate process.

2. Template values.yml itself (if that's even possible in helm), or otherwise embed information from git. I know git itself doesn't really natively support "annotating" files with commit information a la SVN, but it does have "smudge and clean". This solution seems a bit hackish.

3. Move forward with productizing pelorus more, as we've talked about-- release containers and/or charts instead of having folks deploy from the repo itself. While this is a nice "formal" solution, there's value in making it easy for folks to deploy experimental / local versions of exporters as we continue to build out features.

**Comment by etsauer** [comment]
Yeah, a stable branch fixes it for now. The more permanent solution that I
had in mind was to build an action workflow that creates a release commit
to a temporary branch that basically does a search and replace of refs to
master with the version # and then cuts the release from that.
--
Eric Sauer
***@***.***
m:773.318.3036

**Comment by etsauer** [comment]
We could do that either instead of or in conjunction with #3 from your list
--
Eric Sauer
***@***.***
m:773.318.3036

**Comment by etsauer** [comment]
Yes a stable branch fixes the manual process for now.

The more permanent solution that I had in mind was to build an action workflow that creates a release commit to a temporary branch that basically does a search and replace of refs to master/latest with the version # and then cuts a release from that.

We could do that either instead of or in conjunction with #3 from your list

**Comment by KevinMGranger** [comment]
We have identified the issue and mostly solved it. It should be included in the 1.4 release. More details can be found in #329, but for now I'll close this as a duplicate.
